ffs/scheduler/scheduler_import.go

```diff
@@ -0,0 +1,124 @@
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/textileio/powergate/ffs"
+	"github.com/textileio/powergate/ffs/scheduler/internal/cistore"
+)
+
+// ImportDeals create or augments a Storageinfo for a Cid with the provided deal ids.
+// All deal ids must be active on-chain to let the operation succeed.
+func (s *Scheduler) ImportDeals(iid ffs.APIID, payloadCid cid.Cid, dealIDs []uint64) error {
+	// 1. Get current StorageInfo.
+	si, err := s.cis.Get(iid, payloadCid)
+	if err == cistore.ErrNotFound {
+		si = ffs.StorageInfo{
+			APIID:   iid,
+			JobID:   ffs.EmptyJobID,
+			Cid:     payloadCid,
+			Created: time.Now(),
+			Cold: ffs.ColdInfo{
+				Filecoin: ffs.FilInfo{
+					DataCid: payloadCid,
+				},
+			},
+		}
+	} else if err != nil {
+		return fmt.Errorf("getting current storageinfo: %s", err)
+	}
```

Comment:
First, we get the current _StorageInfo_.

If there isn't one, the user is pushing a _StorageConfig_ for the first time, since we don't have any current _StorageInfo_. In this case, we create an empty one with minimal necessary information.

If there is one, the user is augmenting an existing storage with external deals.